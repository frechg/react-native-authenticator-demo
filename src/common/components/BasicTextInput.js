import * as React from 'react';
import { View, Text, TextInput } from 'react-native';
import { formStyles } from '../styles';

export const BasicTextInput = (props) => {
  const {
    field: { name, onBlur, onChange, value },
    form: { errors, touched },
    ...inputProps
  } = props

  const label = props.label ? props.label : false

  return (
    <View style={formStyles.fieldWrapper}>
      { label && <Text style={formStyles.label}>{label}</Text>}
      <TextInput
        style={formStyles.textInput}
        onChangeText={onChange(name)}
        onBlur={onBlur(name)}
        value={value}
        {...inputProps}
      />
      {( errors[name] && touched[name]) && <Text style={formStyles.formError}>{errors[name]}</Text>}
    </View>
  )
}