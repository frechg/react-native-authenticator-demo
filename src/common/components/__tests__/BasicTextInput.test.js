import * as React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react-native';
import { useFormikContext } from 'formik';
import { BasicTextInput } from '../BasicTextInput';

jest.mock('formik');

describe('<BasicTextInput />', () => {
  test('Renders a basic text input', () => {
    const field = {name: 'email', onBlur: jest.fn(), onChange: jest.fn(), value: ''};
    const form = {errors: {}, touched: {}};

    render(
      <BasicTextInput field={field} form={form} />
    )

    expect(screen).toMatchSnapshot();
  });

  test('Renders a basic text input with label', () => {
    const field = {name: 'email', onBlur: jest.fn(), onChange: jest.fn(), value: ''};
    const form = {errors: {}, touched: {}};

    render(
      <BasicTextInput field={field} form={form} label='Email' />
    )

    expect(screen.getByText('Email')).toBeDefined();
    expect(screen).toMatchSnapshot();
  });

  test('Renders a basic text input with default value', () => {
    const email = 'example@email.com';
    const field = {name: 'email', onBlur: jest.fn(), onChange: jest.fn(), value: email};
    const form = {errors: {}, touched: {}};

    render(
      <BasicTextInput field={field} form={form} />
    )

    expect(screen.getByDisplayValue(email)).toBeDefined();
  });

  test('Displays field errors after touched once', () => {
    const field = {name: 'email', onBlur: jest.fn(), onChange: jest.fn(), value: ''};
    const form = {errors: {email: 'Error'}, touched: {email: true}};

    render(
      <BasicTextInput field={field} form={form} />
    )

    expect(screen.getByText('Error')).toBeDefined();
  });

  test('Does not show field error before first touch', () => {
    const field = {name: 'email', onBlur: jest.fn(), onChange: jest.fn(), value: ''};
    const form = {errors: {email: 'Error'}, touched: {email: false}};

    render(
      <BasicTextInput field={field} form={form} />
    )

    expect(screen.queryByText('Error')).toBeFalsy();
  })
});