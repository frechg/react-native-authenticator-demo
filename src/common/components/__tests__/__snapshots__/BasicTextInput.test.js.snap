// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasicTextInput /> Renders a basic text input 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "marginBottom": 24,
      "width": "100%",
    }
  }
>
  <TextInput
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "rgb(140,140,140)",
        "borderRadius": 0,
        "color": "rgb(60,60,60)",
        "fontSize": 18,
        "paddingBottom": 12,
        "paddingRight": 12,
        "paddingTop": 12,
        "width": "100%",
      }
    }
    value=""
  />
</View>
`;

exports[`<BasicTextInput /> Renders a basic text input with label 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "marginBottom": 24,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "rgb(150,150,150)",
        "fontWeight": "bold",
        "marginBottom": 6,
        "textTransform": "uppercase",
      }
    }
  >
    Email
  </Text>
  <TextInput
    label="Email"
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "rgb(140,140,140)",
        "borderRadius": 0,
        "color": "rgb(60,60,60)",
        "fontSize": 18,
        "paddingBottom": 12,
        "paddingRight": 12,
        "paddingTop": 12,
        "width": "100%",
      }
    }
    value=""
  />
</View>
`;
